src/main/java/picard/analysis/CollectQualityYieldMetrics.java

```diff
@@ -77,74 +79,112 @@
-    public boolean INCLUDE_SECONDARY_ALIGNMENTS = false;
+    public boolean INCLUDE_SECONDARY_ALIGNMENTS = this.collector.includeSecondaryAlignments;
 
     @Option(doc="If true, include bases from supplemental alignments in metrics. Setting to true may cause double-counting " +
             "of bases if there are supplemental alignments in the input file.")
-    public boolean INCLUDE_SUPPLEMENTAL_ALIGNMENTS = false;
-
-    // The metrics to be accumulated
-    private final QualityYieldMetrics metrics = new QualityYieldMetrics();
+    public boolean INCLUDE_SUPPLEMENTAL_ALIGNMENTS = this.collector.includeSupplementalAlignments;
 
     /** Ensure that we get all reads regardless of alignment status. */
     @Override protected boolean usesNoRefReads() { return true; }
 
     @Override
     protected void setup(final SAMFileHeader header, final File samFile) {
         IOUtil.assertFileIsWritable(OUTPUT);
+        this.collector = new QualityYieldMetricsCollector(USE_ORIGINAL_QUALITIES, INCLUDE_SECONDARY_ALIGNMENTS, INCLUDE_SUPPLEMENTAL_ALIGNMENTS);
     }
 
     @Override
     protected void acceptRead(final SAMRecord rec, final ReferenceSequence ref) {
-        if (!INCLUDE_SECONDARY_ALIGNMENTS    && rec.getNotPrimaryAlignmentFlag()) return;
-        if (!INCLUDE_SUPPLEMENTAL_ALIGNMENTS && rec.getSupplementaryAlignmentFlag()) return;
+        this.collector.acceptRecord(rec, ref);
+    }
+
+    @Override
+    protected void finish() {
+        final MetricsFile<QualityYieldMetrics, Integer> metricsFile = getMetricsFile();
+        this.collector.finish();
+        this.collector.addMetricsToFile(metricsFile);
+        metricsFile.write(OUTPUT);
+    }
+
+    public static class QualityYieldMetricsCollector {
+        // If true, include bases from secondary alignments in metrics. Setting to true may cause double-counting
+        // of bases if there are secondary alignments in the input file.
+        private final boolean useOriginalQualities;
+
+        // If true, include bases from secondary alignments in metrics. Setting to true may cause double-counting
+        // of bases if there are secondary alignments in the input file.
+        private final boolean includeSecondaryAlignments;
+
+        // If true, include bases from supplemental alignments in metrics. Setting to true may cause double-counting
+        // of bases if there are supplemental alignments in the input file.
+        public final boolean includeSupplementalAlignments;
 
-        final int length = rec.getReadLength();
-        metrics.TOTAL_READS++;
-        metrics.TOTAL_BASES += length;
+        // The metrics to be accumulated
+        private final QualityYieldMetrics metrics = new QualityYieldMetrics();
 
-        final boolean isPfRead = !rec.getReadFailsVendorQualityCheckFlag();
-        if (isPfRead) {
-            metrics.PF_READS++;
-            metrics.PF_BASES += length;
+        public QualityYieldMetricsCollector() {
```

Comment:
See my comment above about storing the defaults in one place.   Again, I can make the defaults their own variables that both `CollectQualityYieldMetrics ` and `QualityYieldMetricsCollector` reference.